assets/src/gaussholder.js

```diff
@@ -144,26 +144,22 @@ export default function () {
 	 * Render all placeholders on the page
 	 */
 	return function () {
-		let images = document.getElementsByTagName( 'img' );
-
-		for ( let i = images.length - 1; i >= 0; i-- ) {
-			let img = images[i];
-
-			// Ensure the blank GIF has loaded first
-			if ( img.complete ) {
-				handleElement( img );
-			} else {
-				img.onload = function () {
-					handleElement( this );
-				};
-			}
-		}
+		const images = document.getElementsByTagName( 'img' );
+
+		const options = {
+			rootMargin: '1200px'
+		};
 
-		loadLazily = images;
-		scrollHandler();
+		const imagesObserver = new IntersectionObserver( (entries) => {
+			const visibleImages = entries
+				.filter( ( { isIntersecting } ) => isIntersecting === true );
 
-		if ( loadLazily.length > 0 ) {
-			window.addEventListener( 'scroll', scrollHandler );
-		}
+			visibleImages.forEach( ( { target } ) => loadOriginal( target ) );
```

Comment:
It would be good to unobserve the IntersectionObserver here (or as part of the `loadOriginal()` function), since it won't be needed after the image is loaded, and clean up any extra observers running in the background.